utility.hpp.in

```diff
@@ -13,17 +13,28 @@
  */
 namespace utility{
 
-/* forward Declaration of function signatures */
+/* functions */
+/** @returns A copy of the argument value */
+template< typename T >
+T   copy( const T& t ){ return t; }
```

Comment:
This is essentially the copy equivalent of the std::move function. I assume we'll be using this when we do not want to move an argument into an rvalue reference?

I am going to be the advocate of the devil here: if we use this too often, would it not be better to not use rvalue references (using a local variable instead) in a function call and explicitly use std::move when we want to move instead of copy? I think this is the more standard way (from what I see on the internet, I would assume this is the case) and as such it would be easier for other people to understand the code.